lib/ansible/modules/net_tools/netplan.py

```diff
@@ -0,0 +1,903 @@
+        default setting.
+    required: false
+    type: bool
+
+  addresses:
+    description:
+      - Add static addresses to the interface in addition to the ones received
+        through DHCP or RA. Each sequence entry is in CIDR notation, i. e. of
+        the form C(addr)/C(prefixlen). C(addr) is an IPv4 or IPv6 address as
+        recognized by C(inet_pton) and C(prefixlen) the number of bits of the
+        subnet.
+    required: false
+    type: list
+
+  gateway4:
+    description:
+      - Set default gateway for IPv4, for manual address configuration.
+        This requires setting addresses too. Gateway IPs must be in a form
+        recognized by C(inet_pton).
+    required: false
+
+  gateway6:
+    description:
+      - Set default gateway for IPv6, for manual address configuration.
+        This requires setting addresses too. Gateway IPs must be in a form
+        recognized by C(inet_pton).
+    required: false
+
+  nameservers-search:
+    description:
+      -  Set a list of DNS servers search domains.
+    required: false
+    type: list
+
+  nameservers-addresses:
+    description:
+      -  Set a list of ipv4/ipv6 DNS servers.
+    required: false
+    type: list
+
+  macaddress:
+    description:
+      - Set the device's MAC address. The MAC address must be in the form
+        XX:XX:XX:XX:XX:XX.
+    required: false
+
+  mtu:
+    description:
+      - Set the Maximum Transmission Unit for the interface.
+        The default is 1500. Valid values depend on your network interface.
+    required: false
+    type: int
+
+  optional:
+    description:
+      - An optional device is not required for booting. Normally, networkd
+        will wait some time for device to become configured before proceeding
+        with booting. However, if a device is marked as optional, networkd will
+        not wait for it. This is only supported by networkd, and the
+        default is false.
+    required: false
+    type: bool
+
+  optional-addresses:
+    description:
+      - Specify types of addresses that are not required for a device to be
+        considered online. This changes the behavior of backends at boot time
+        to avoid waiting for addresses that are marked optional, and thus
+        consider the interface as "usable" sooner. This does not disable
+        these addresses, which will be brought up anyway.
+    required: false
+    type: list
+
+  match-name:
+    description:
+      - This selects the current interface name in physical devices
+        by various hardware properties.
+    required: false
+
+  match-macaddress:
+    description:
+      - This selects the physical device that matchs with MAC address
+        in the form "XX:XX:XX:XX:XX:XX".
+    required: false
+
+  match-driver:
+    description:
+      - This selects the physical device that matchs with Kernel driver name
+        corresponding to the C(DRIVER) udev property.
+    required: false
+
+  set-name:
```

Comment:
I'm unhappy with the parameter name (sic!), but I can't find a much better option offhand.
Perhaps simply `name`?